public/app/features/dashboard-scene/scene/NavToolbarActions.tsx

```diff
@@ -126,6 +136,19 @@ export const NavToolbarActions = React.memo<Props>(({ dashboard }) => {
         </Button>
       );
     }
+
+    if (meta.isSnapshot) {
+      toolbarActions.push(
+        <DashNavButton
+          key="button-snapshot"
+          tooltip={t('dashboard.toolbar.open-original', 'Open original dashboard')}
+          icon="link"
+          onClick={() => {
+            dashboard.onOpenSnapshotOriginalDashboard();
+          }}
+        />
```

Comment:
I would put this in a new GoToSnapshotOriginButton react component and put the logic of onOpenSnapshotOriginalDashboard inside it (to not pollute DashboardScene with it)

Reply:
I will do it in another PR so I avoid future conflicts, these files are changing heavily.